crates/solver/src/driver.rs

```diff
@@ -630,6 +634,74 @@ fn orders_with_price_estimates(
     orders
 }
 
+// In case we cannot estimate the price for a token but there is only one pair traded in the batch the accuracy of the price estimate for ranking a solution doesn't matter that much (surplus is not competing with other tokens).
+// Therefore we can deduct a price estimate by looking at the average limit price of the orders on the single token pair
+fn augment_prices(orders: &[LimitOrder], prices: &mut HashMap<H160, BigRational>) {
+    if orders.is_empty() {
+        return;
+    }
+    let first_order = orders.first().unwrap();
+    let init = (
+        (first_order.sell_token, U256::zero()),
+        (first_order.buy_token, U256::zero()),
+    );
+    let only_pair_with_amounts = orders.iter().try_fold(
+        init,
+        |(first_token_and_amount, second_token_and_amount), order| {
```

Comment:
ubernit: You can double de-structure here to avoid all the `.0` and `.1`:
```suggestion
        |((first_token, first_amount), (second_token, second_amount)), order| {
```

If you you find the line too long you can also split it:
```suggestion
        |(first_token_and_amount, second_token_and_amount), order| {
            let (first_token, first_amount) = first_token_and_amount;
            let (second_token, second_amount) = second_token_and_amount;
```